src/object/obj_class.c

```diff
@@ -234,9 +234,24 @@ daos_oclass_grp_nr(struct daos_oclass_attr *oc_attr, struct daos_obj_md *md)
 	return oc_attr->ca_grp_nr;
 }
 
+/**
+ * To honor RF setting during failure cases, let's reserve RF
+ * groups, so if some targets fail, there will be enough replacement
+ * targets to rebuild, so to avoid putting multiple shards in the same
+ * domain, which may break the RF setting.
+ *
+ * Though let's keep reserve targets to be less than 30% of the total
```

Comment:
can you please add some comments here as to why 30% max?
what happens if rf > 30% (probably unlikely though to run into this) ?

Reply:
30% is just a rough number to avoid reserving too much targets here.   So we will reserve at most 30% of the total targets, if rf > 30%, then it will not reserve too much targets for the objects, i.e. RF promise might be broken even before RF targets are excluded.

Anyway,  since this patch is just the temporary solution for this RF dilemma for 2.4, so we are are trying to balance the performance and RF.